core/src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -310,6 +310,83 @@ class GraphFrame private (
       .agg(count("*").cast("int").as("degree"))
   }
 
+  /**
+   * The out-degree of each vertex per edge type, returned as a DataFrame with two columns:
+   *   - [[GraphFrame.ID]] the ID of the vertex
+   *   - "outDegrees" a struct with a field for each edge type, storing the out-degree count
+   *
+   * @param edgeTypeCol
+   *   Name of the column in edges DataFrame that contains edge types
+   * @group degree
+   */
+  def typeOutDegree(edgeTypeCol: String): DataFrame = {
```

Comment:
Let's allow users to pass a known list of values of the `edgeTypeCol`? It will benefit both pivot and case-when approach.